runtime/stdlib/test_emulatorbackend.go

```diff
@@ -494,6 +533,129 @@ func (t *testEmulatorBackendType) newUseConfigFunction(testFramework TestFramewo
+
+const testEmulatorBackendTypeLogsFunctionDocString = `
+Returns all the logs from the blockchain, up to the calling point.
+`
+
+func (t *testEmulatorBackendType) newLogsFunction(
+	testFramework TestFramework,
+) *interpreter.HostFunctionValue {
+	return interpreter.NewUnmeteredHostFunctionValue(
+		t.logsFunctionType,
+		func(invocation interpreter.Invocation) interpreter.Value {
+			logs := testFramework.Logs()
+			inter := invocation.Interpreter
+
+			arrayType := interpreter.NewVariableSizedStaticType(
+				inter,
+				interpreter.NewPrimitiveStaticType(
+					inter,
+					interpreter.PrimitiveStaticTypeString,
+				),
+			)
+
+			values := make([]interpreter.Value, len(logs))
+			for i, log := range logs {
+				memoryUsage := common.NewStringMemoryUsage(len(log))
+				values[i] = interpreter.NewStringValue(
+					inter,
+					memoryUsage,
+					func() string {
+						return log
+					},
+				)
+			}
+
+			return interpreter.NewArrayValue(
+				inter,
+				invocation.LocationRange,
+				arrayType,
+				common.ZeroAddress,
+				values...,
+			)
+		},
+	)
+}
+
+// 'EmulatorBackend.serviceAccount' function
+
+const testEmulatorBackendTypeServiceAccountFunctionName = "serviceAccount"
+
+const testEmulatorBackendTypeServiceAccountFunctionDocString = `
+Returns the service account of the blockchain. Can be used to sign
+transactions with this account.
+`
+
+func (t *testEmulatorBackendType) newServiceAccountFunction(
+	testFramework TestFramework,
+) *interpreter.HostFunctionValue {
+	return interpreter.NewUnmeteredHostFunctionValue(
+		t.serviceAccountFunctionType,
+		func(invocation interpreter.Invocation) interpreter.Value {
+			serviceAccount, err := testFramework.ServiceAccount()
+			if err != nil {
+				panic(err)
+			}
+
+			return newTestAccountValue(
+				testFramework,
+				invocation.Interpreter,
+				invocation.LocationRange,
+				serviceAccount,
+			)
+		},
+	)
+}
+
+// 'EmulatorBackend.getEvents' function
+
+const testEmulatorBackendTypeGetEventsFunctionName = "getEvents"
+
+const testEmulatorBackendTypeGetEventsFunctionDocString = `
+Returns all events emitted from the blockchain,
+optionally filtered by event type.
+`
+
+func (t *testEmulatorBackendType) newGetEventsFunction(
+	testFramework TestFramework,
+) *interpreter.HostFunctionValue {
+	return interpreter.NewUnmeteredHostFunctionValue(
+		t.getEventsFunctionType,
+		func(invocation interpreter.Invocation) interpreter.Value {
+			eventType, ok := invocation.Arguments[0].(*interpreter.StringValue)
+			if !ok {
+				panic(errors.NewUnreachableError())
+			}
+			events := testFramework.GetEvents(eventType.Str)
+
+			inter := invocation.Interpreter
+
+			arrayType := interpreter.NewVariableSizedStaticType(
+				inter,
+				interpreter.PrimitiveStaticTypeAnyStruct,
+			)
+
+			values := make([]interpreter.Value, len(events))
+			// for i, event := range events {
+			//  How do I convert a cadence.Event to interpreter.Value ?
+			// 	values[i] = event
+			// }
+
```

Comment:
As for the conversion, we can return the `interpreter.Value` from the `testFramework.GetEvents()`. and have the conversion there.

This is also how the `RunScript` works currently: https://github.com/onflow/cadence-tools/blob/8ebd4b9835c4ad46bd28ece858292e367ece137a/test/emulator_backend.go#L167-L176. Probably don't need to return the errors like in scripts, returning just a `[]interpreter.Value` would be sufficient.

Reply:
You are right. I think this is the only way to go about it, because I ran into some cyclic import errors, when trying to use `runtime.ImportValue` inside `stdlib` package (which already imports `runtime` package)

Reply:
I have made some changes in that direction, in this commit: https://github.com/onflow/cadence/pull/2505/commits/1a52bef821c72162cd4f3d3b216ce0a9312cd604. I will update accordingly the PR https://github.com/onflow/cadence-tools/pull/113. I am still not sure how to wrap a `[]interpreter.Value` inside a `interpreter.Value`, I will give it a shot though.

Reply:
Updated in https://github.com/onflow/cadence-tools/pull/113/commits/72eee50a3699d94abcc474141d6eec48668dfbcc